### ISSUE TITLE
Using float datatype features with hough_train.py

### ISSUE THREAD

**Issue by priyankavokuda** [classner/fertilized-forests#43]
Hi,

I am trying to run python example hough_train.py using deep features instead of features given in the example.
The deep features are of datatype float32.

I get following error:
` NoCopyPatchSampleManager
    patches_luc
ValueError: Object at sequence position 0 can not be converted to target type N7ndarray5ArrayIKhLi3ELi3EEE!
` 
Example of what I am trying to do: 
`feat_images = [np.random.randint(0, 500, size=(128, im.shape[0], im.shape[1])).astype(np.float) for im in cvimages]`

I also observed that possible instantiations for arguments of method NoCopyPatchSampleManager are:
   * - uint8_t; int16_t
   * - uint8_t; uint

Is there a way to use features of datatype float32?

**Comment by classner** [comment]
Hi,

You perfectly identified the reason for the problem: the sample manager is not instantiated for the datatype you want to use. As far as I understand, you need `uint8_t;float32`, right? You can just add  this line to the comment [here](https://github.com/classner/fertilized-forests/blob/master/include/fertilized/data_providers/nocopypatchsamplemanager.h#L32), clone the [fertilized-devtools](https://github.com/classner/fertilized-devtools) package and run it to generate the interface code. When you rebuild, the new interface is available.

This is a bit of an overhead, but it was at the time the most reliable solution to keep three interfaces up-to-date... :-/

Best,
Christoph

**Comment by priyankavokuda** [comment]
Hi Christoph,

Thank you for your quick reply :)

I need `float32,int16_t` as first two arguments for `NoCopyPatchSampleManager ` function.
I have added line `* - float32; int16_t` in `nocopypatchsamplemanager.h` and its parent classes (```isamplemanager.h
patchsamplemanager.h```).
after executing generate.h, when I rebuild(during cmake --build . ) I get the following error:
1.
`In file included from fertilized-forests/include/fertilized/./global.h:6:0,
                 from fertilized-forests/include/fertilized/fertilized.h:33,
                 from fertilized-forests/tests/boosting.cpp:2:
fertilized-forests/include/fertilized/version.h:2:27: note: ‘const unsigned int FERTILIZED_LIB_VERSION()’ previously defined here
 static const unsigned int FERTILIZED_LIB_VERSION() { return 102U; }`

and 
2.

`In file included from fertilized-forests/bindings/python/exporters/__264_exporter.cpp:6:0:
fertilized-forests/bindings/python/exporters/./__StandardFeatureSelectionProvider_exporter.h: In function ‘void pyfertilized::export_StandardFeatureSelectionProvider(const string&)’:
fertilized-forests/bindings/python/exporters/./__StandardFeatureSelectionProvider_exporter.h:91:28: error: type/value mismatch at argument 4 in template parameter list for ‘template<class T0, class T1, class T2, class T3, class T4, class T5, class T6, class T7, class T8, class T9, class T10, class T11, class T12, class T13, class T14> class boost::python::init’
                          >())
                            ^
fertilized-forests/bindings/python/exporters/./__StandardFeatureSelectionProvider_exporter.h:91:28: note:   expected a type, got ‘boost::python::optional<long unsigned int, boost::python::optional<const unsigned int&> >()’
bindings/python/CMakeFiles/pyfertilized_5.dir/build.make:398: recipe for target 'bindings/python/CMakeFiles/pyfertilized_5.dir/exporters/__264_exporter.cpp.o' failed`

Similar error also in `ClassificationThresholdOptimizer_exporter.h`

Thank you,
Priyanka

**Comment by classner** [comment]
Hi Priyanka,

it sounds like there may still be build artifacts around that lead to problems. Have you tried a fresh build after you regenerated the interface files (i.e., remove the build folder and start the build from scratch)?

Christoph

**Comment by priyankavokuda** [comment]
Hi Christoph,

Yes, this is after a fresh build.

**Comment by classner** [comment]
Hi Priyanka,

not sure what causes the first error; it is because this function appears in two places. Maybe you can track it down using your editor's search or with a `grep` command. Let me know what you find out!

The second: try deleting `bindings/python/exporters/*` as well as other re-generated folders. I assume you removed other classes or class types, then not all of the files will be properly overwritten.

Best,
Christoph

**Comment by priyankavokuda** [comment]
Hi Christoph,

update on the errors:

- 1st error:  was caused by artifacts from failed build process.

- 2nd error:  the error still remains during `cmake --build` after executing generate.py. 

I have deleted `bindings/python/exporters/*` and other regenerated files or folders in `bindings` folder (regenerated by generate.py).

(repeating) for the requirement of float input_dtype I've added line `* - float32; int16_t ` in following files:
```
nocopypatchsamplemanager.h
patchsamplemanager.h
isamplemanager.h
sample.h
```

and 
>  I assume you removed other classes or class types, then not all of the files will be properly overwritten.

I not sure which classes or class types you are referring here.

Thank you,
Priyanka
